components/server/ee/src/billing/entitlement-service-license.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { UserDB } from "@gitpod/gitpod-db/lib";
+import {
+    User,
+    WorkspaceInstance,
+    WorkspaceTimeoutDuration,
+    WORKSPACE_TIMEOUT_DEFAULT_LONG,
+    WORKSPACE_TIMEOUT_DEFAULT_SHORT,
+} from "@gitpod/gitpod-protocol";
+import { LicenseEvaluator } from "@gitpod/licensor/lib";
+import { Feature } from "@gitpod/licensor/lib/api";
+import { inject, injectable } from "inversify";
+import { EntitlementService } from "../../../src/billing/entitlement-service";
+import { Config } from "../../../src/config";
+import { MayStartWorkspaceResult } from "../user/eligibility-service";
+
+@injectable()
+export class EntitlementServiceLicense implements EntitlementService {
+    @inject(Config) protected readonly config: Config;
+    @inject(UserDB) protected readonly userDb: UserDB;
+    @inject(LicenseEvaluator) protected readonly licenseEvaluator: LicenseEvaluator;
+
+    async mayStartWorkspace(
+        user: User,
+        date: Date,
+        runningInstances: Promise<WorkspaceInstance[]>,
+    ): Promise<MayStartWorkspaceResult> {
+        // if payment is not enabled users can start as many parallel workspaces as they want
+        return { enoughCredits: true };
+    }
+
+    async maySetTimeout(user: User, date: Date): Promise<boolean> {
+        // when payment is disabled users can do everything
+        return true;
+    }
+
+    async getDefaultWorkspaceTimeout(user: User, date: Date): Promise<WorkspaceTimeoutDuration> {
+        const userCount = await this.userDb.getUserCount(true);
+
+        // the self-hosted case
+        if (!this.licenseEvaluator.isEnabled(Feature.FeatureSetTimeout, userCount)) {
+            return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
+        }
+
+        return WORKSPACE_TIMEOUT_DEFAULT_LONG;
+    }
+
+    async userGetsMoreResources(user: User): Promise<boolean> {
+        // TODO(gpl) Not sure this makes sense, but it's the way it was before
+        return false;
```

Comment:
`return true` actually cf. https://github.com/gitpod-io/gitpod/pull/11889/files#diff-da9747280402a215ed0c5b3a2a4cf352032977892ab33a1686c6b0f73019cef2L175

Reply:
Ya-ha, this is where it's getting creepy :grimacing: Thank you for the thorough review, exactly what I hoped for! :pray: 

This methods only direct callsite was [here](https://github.com/gitpod-io/gitpod/pull/11889/files#diff-6eac038c4219e90db377c71da65f9a2a5521b0ad013330d63d8b0ff6f10aa65aL76), which returns `false`... again, not sure this is the best decision, but want to strictly keep existing behavior in this PR.

Reply:
dead short